File: src/main/java/net/atos/zac/documenten/model/OntkoppeldDocumentListParameters.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.documenten.model;

import net.atos.zac.shared.model.ListParameters;
import net.atos.zac.zoeken.model.DatumRange;


public class OntkoppeldDocumentListParameters extends ListParameters {

    private String titel;

    private DatumRange creatiedatum;

    private String zaakID;

    private String ontkoppeldDoor;

    private DatumRange ontkoppeldOp;

    private String reden;

    public OntkoppeldDocumentListParameters() {
        super();
    }

    public String getTitel() {
        return titel;
    }

    public void setTitel(final String titel) {
        this.titel = titel;
    }

    public String getReden() {
        return reden;
    }

    public void setReden(final String reden) {
        this.reden = reden;
    }

    public DatumRange getCreatiedatum() {
        return creatiedatum;
    }

    public void setCreatiedatum(final DatumRange creatiedatum) {
        this.creatiedatum = creatiedatum;
    }

    public String getOntkoppeldDoor() {
        return ontkoppeldDoor;
    }

    public void setOntkoppeldDoor(final String ontkoppeldDoor) {
        this.ontkoppeldDoor = ontkoppeldDoor;
    }

    public String getZaakID() {
        return zaakID;
    }

    public void setZaakID(final String zaakID) {
        this.zaakID = zaakID;
    }

    public DatumRange getOntkoppeldOp() {
        return ontkoppeldOp;
    }

    public void setOntkoppeldOp(final DatumRange ontkoppeldOp) {
        this.ontkoppeldOp = ontkoppeldOp;
    }
}

File: src/main/app/src/app/documenten/model/ontkoppeld-document-list-parameters.ts
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

import { SortDirection } from "@angular/material/sort";
import { ListParameters } from "../../shared/model/list-parameters";
import { User } from "../../identity/model/user";
import { DatumRange } from "../../zoeken/model/datum-range";
import { ZoekFilters } from "../../gebruikersvoorkeuren/zoekopdracht/zoekopdracht.component";

export class OntkoppeldDocumentListParameters
  extends ListParameters
  implements ZoekFilters
{
  readonly filtersType = "OntkoppeldDocumentListParameters";
  zaakID: string;
  ontkoppeldDoor: User;
  ontkoppeldOp = new DatumRange();
  creatiedatum = new DatumRange();
  titel: string;
  reden: string;

  constructor(sort: string, order: SortDirection) {
    super(sort, order);
  }

  static heeftActieveFilters(zoekFilters: any): boolean {
    if (zoekFilters.zaakID != null) {
      return true;
    }
    if (zoekFilters.ontkoppeldDoor != null) {
      return true;
    }
    if (
      zoekFilters.ontkoppeldOp?.van != null ||
      zoekFilters.ontkoppeldOp?.tot != null
    ) {
      return true;
    }
    if (
      zoekFilters.creatiedatum?.van != null ||
      zoekFilters.creatiedatum?.tot != null
    ) {
      return true;
    }
    if (zoekFilters.titel != null) {
      return true;
    }
    if (zoekFilters.reden != null) {
      return true;
    }
    return false;
  }
}
